brain/compartmentReportView.cpp

```diff
@@ -120,9 +56,23 @@ const CompartmentReportMapping& CompartmentReportView::getMapping() const
 }
 
 std::future<CompartmentReportFrame> CompartmentReportView::load(
-    const float timestamp)
+    double timestamp)
 {
+    const double start = _impl->report->getStartTime();
+    const double end = _impl->report->getEndTime();
+
+    if (timestamp < start || timestamp > end)
+        throw std::logic_error("Invalid timestamp");
+
+    const double timestep = _impl->report->getTimestep();
+
+    size_t frameIndex = std::min(size_t(round((end - start) / timestep)) - 1,
+                                 size_t(round(timestamp / timestep)));
+
+    timestamp = start + frameIndex * timestep;
```

Comment:
You already know timestamp is in [start, end) so replace lines 69-72 with:
timestamp = start + std::floor((timestamp - start) / timestep) * timestep;